src/pages/blog/airship2-under-the-hood-lifecycle.md

```diff
@@ -0,0 +1,666 @@
+
+Plans can also share phases, which makes them another fairly light-weight construct and allows YAML engineers to craft any number of specific plans without duplicating plan definitions.
+
+## Defining a Plan
+
+Below is an example of an Airship PhasePlan document. The phases section describes an ordered list of phase names that have been declared in their own documents and this plan if executed will run each of the phases defined.
+
+```yaml
+---
+apiVersion: airshipit.org/v1alpha1
+kind: PhasePlan
+metadata:
+  name: phasePlan
+description: "Default phase plan"
+phases:
+  - name: initinfra-ephemeral
+  - name: clusterctl-init-ephemeral
+  - name: controlplane-ephemeral
+  - name: clusterctl-init-target
+  - name: clusterctl-move
+  - name: initinfra-target
+  - name: controlplane-target
+  - name: workers-target
+  - name: workload-target
+```
+
+Once plans are defined, you can then interact with them using the `airshipctl plan command. For instance:
+
+```
+airshipctl plan
+
+This command provides capabilities for interacting with plan objects,
+responsible for execution phases in groups
+
+Usage:
+  airshipctl plan [command]
+Available Commands:
+  list        List plans
+  run         Run plan
+
+Flags:
+  -h, --help help for plan
+
+Global Flags:
+      --airshipconf string Path to file for airshipctl configuration. (default "$HOME/.airship/config")
+      --debug enable verbose output
+
+Use "airshipctl plan [command] --help" for more information about a command.
+
+```
+
+For Airship v2, airshipctl plan provides to subcommands :
+  * _**List:**_ Will list the details for the phases defined in the plan.
+  * _**Run:**_ Will run the phases defined by the plan.
+
+For future releases, other subcommands in par with the phase command will be available. Such as validate, render, status, etc.
+
+## The Airship Lifecycle
+
+
+Airship intends to be opinionated about deployments only to the point of enabling the goals we’ve discussed so far--and in fact--seeks to step out of the way as much as possible to let the end-user declarations drive configurations. This is the only way Airship 2.0 can keep up with a constantly evolving CNCF landscape. We have achieved this by leveraging a command line utility [airshipctl](https://github.com/airshipit/airshipctl) that drives the deployment and lifecycle management of Kubernetes clouds.  As explained we articulate lifecycle management as a plan, which allows us to drive the lifecycle as a single button approach. But we have made it simple by making that single button a composite of multiple phases, each driveable on its own explicitly.
+
+
+The image below expresses the lifecycle in terms of Stages and Phases.
+
+![](/img/under_the_hood_blog/4.png)
```

Comment:
I can barely guess what is written on this picture. Is it possible to make a bigger text/picture?